### ISSUE TITLE
Add table `azure_maintenance_configuration` + items

### ISSUE THREAD

**Issue by Shr3ps** [turbot/steampipe-plugin-azure#724]
Azure is now pushing up the new service Update Center.
We would like to have a new table to list all update maintenance configurations in place and also would be great to be able to list all items (vm) attached with their update "state".

SDK model I found: https://github.com/Azure/azure-sdk-for-go/blob/v58.0.0/services/maintenance/mgmt/2021-05-01/maintenance/models.go

![image](https://github.com/turbot/steampipe-plugin-azure/assets/1845672/fa4e8cf9-8232-4cac-b5d8-a8feaa71c43c)

**Comment by ParthaI** [comment]
Hello @Shr3ps, thank you for bringing this issue to our attention.

I've raised a [PR](https://github.com/turbot/steampipe-plugin-azure/pull/725) to introduce a new table, `azure_maintenance_configuration` which will provide us the information about the Maintenance Configuration.

Currently, there is no direct API to fetch the update status of VMs along with the count of critical, missing, and other updates as depicted in the snapshot provided in the issue description.

I came across [this article](https://learn.microsoft.com/en-us/answers/questions/475393/azure-update-management-data-retrieve-via-rest-api) which, unfortunately, doesn't fully meet our needs.

We plan to revisit this once Azure extends their API support to cover these requirements.

Thank you!
